### ISSUE TITLE
Internal server error after upgrade to nextcloud 18

### ISSUE THREAD

**Issue by leArthurDent** [nextcloud/mail#2747]
### Expected behavior
Nextcloud opens Mail app

### Actual behavior

> Interner Serverfehler
>
> Der Server konnte die Anfrage nicht fertig stellen.
>
> Sollte dies erneut auftreten, sende bitte die nachfolgenden technischen Einzelheiten an Deinen > Server-Administrator.
>
> Weitere Details können im Server-Protokoll gefunden werden.
> Technische Details
>
> Entfernte Adresse: xx.xxx.xx.xxx
> Anfragekennung: xxxxxxxxxxxxx

Logmessage:

[index] Error: Exception: explode() expects parameter 1 to be string, null given at <<closure>>

0. /var/www/html/nextcloud/lib/private/AppFramework/App.php line 125
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Mail\Controller\PageController {}, "index")
1. /var/www/html/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   OC\AppFramework\App::main("OCA\\Mail\\Controller\\PageController", "index", OC\AppFramework\ ... {}, {_route: "mail.page.index"})
2. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({_route: "mail.page.index"})
3. /var/www/html/nextcloud/lib/private/Route/Router.php line 299
   call_user_func(OC\AppFramework\ ... {}, {_route: "mail.page.index"})
4. /var/www/html/nextcloud/lib/base.php line 1008
   OC\Route\Router->match("/apps/mail/")
5. /var/www/html/nextcloud/index.php line 38
   OC::handleRequest()

### Mail app

**Mail app version:** 1.1.3

**Mailserver or service:** (Posteo)

### Server configuration

**Operating system**: Raspbian GNU/Linux 10 (buster)

**Web server:** Apache/2.4.38 (Raspbian)

**Database:** 10.3.22-MariaDB-0+deb10u1 Raspbian 10

**PHP version:** PHP 7.3.14-1~deb10u1

**Nextcloud Version:** 18.0.2

#### Client configuration
**Browser:** Firefox 68.5.0esr (64-Bit)

**Operating system:** openSUSE 15.1

**Comment by in23o** [comment]
Same here since 18.0 

Webserver: Apache (cgi-fcgi)
Database: mysql 5.6.47
PHP version: 7.2.5

Modules loaded: Core, date, libxml, openssl, pcre, zlib, bcmath, bz2, calendar, ctype, curl, dom, hash, fileinfo, filter, ftp, gd, gettext, gmp, SPL, iconv, session, intl, json, ldap, mbstring, standard, mysqlnd, odbc, pcntl, mysqli, PDO, pdo_mysql, Phar, posix, pspell, Reflection, imap, shmop, SimpleXML, snmp, soap, sockets, exif, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, cgi-fcgi

Nextcloud version: 18.0.2 - 18.0.2.2

Enabled:
 - accessibility: 1.4.0
 - activity: 2.11.0
 - announcementcenter: 3.7.0
 - bruteforcesettings: 1.5.0
 - calendar: 2.0.2
 - circles: 0.18.3
 - cloud_federation_api: 1.1.0
 - comments: 1.8.0
 - contacts: 3.2.0
 - dav: 1.14.0
 - deck: 0.8.0
 - event_update_notification: 1.0.1
 - federatedfilesharing: 1.8.0
 - federation: 1.8.0
 - files: 1.13.1
 - files_pdfviewer: 1.7.0
 - files_rightclick: 0.15.2
 - files_sharing: 1.10.1
 - files_trackdownloads: 1.7.0
 - files_trashbin: 1.8.0
 - files_versions: 1.11.0
 - files_videoplayer: 1.7.0
 - firstrunwizard: 2.7.0
 - groupfolders: 6.0.1
 - logreader: 2.3.0
 - lookup_server_connector: 1.6.0
 - mail: 1.1.3
 - nextcloud_announcements: 1.7.0
 - notifications: 2.6.0
 - oauth2: 1.6.0
 - onlyoffice: 4.1.4
 - password_policy: 1.8.0
 - photos: 1.0.0
 - privacy: 1.2.0
 - provisioning_api: 1.8.0
 - quota_warning: 1.7.0
 - registration: 0.4.7
 - serverinfo: 1.8.0
 - settings: 1.0.0
 - sharebymail: 1.8.0
 - spreed: 8.0.5
 - support: 1.1.0
 - survey_client: 1.6.0
 - systemtags: 1.8.0
 - terms_of_service: 1.4.0
 - text: 2.0.0
 - theming: 1.9.0
 - theming_customcss: 1.5.0
 - twofactor_backupcodes: 1.7.0
 - updatenotification: 1.8.0
 - video_converter: 0.1.3
 - viewer: 1.2.0
 - workflowengine: 2.0.0
Disabled:
 - admin_audit
 - encryption
 - files_external
 - recommendations
 - user_ldap

**Comment by ChristophWurst** [comment]
@in23o you have a different error than OP

**Comment by leArthurDent** [comment]
Hey there, I was able to work around the server error by deleting all related entries in the database and reinstalling the app as previously used [here](https://help.nextcloud.com/t/how-to-fully-remove-and-reinstall-the-mail-app/59469/10).

After backing up the current database using mysqldump I ran the following commands in mysql:

```
DELETE FROM "oc_migrations" WHERE app="mail";
DELETE FROM "oc_preferences" WHERE appid="mail";

drop table oc_mail_accounts;
drop table oc_mail_attachments;
drop table oc_mail_aliases;
drop table oc_mail_coll_addresses;
drop table oc_mail_mailboxes;
drop table oc_mail_messages;
drop table oc_mail_recipients;
drop table oc_mail_classifier;
```

Judging from the code in FolderMapper.php and the error message, I suggest that the error was caused by ```getDelimiter()``` trying to read the delimiter from some folder class but ended up with NULL.

Since I neither know php, nor am I very well with databases, I decided to solve this error hammer-style.
